### PR TITLE
Use relation in TI join to RTIF

### PR DESCRIPTION
**Pull request by dstandish**

### CODE REVIEW

**Comment by ashb** [comment]
Oh this world use the relation instead of specifying the condition

**Review by ashb — COMMENTED** [review]
Like we do on L81/all the other end points.

**Review by uranusjr — APPROVED** [review]
Static checks need fixing.

**Review by ashb — APPROVED** [review]

File: tests/api_connexion/endpoints/test_task_instance_endpoint.py [hunk 1]
```diff
@@ -293,6 +293,57 @@ def test_should_respond_200_task_instance_with_sla_and_rendered(self, session):
...
+            setattr(new_ti, attr, getattr(ti, attr))
+        session.add(new_ti)
+        rendered_fields = RTIF(new_ti, render_templates=False)
+        session.add(rendered_fields)
+        session.commit()
+
+        # in each loop, we should get the right mapped TI back
+        for map_index in (1, 2):
+            response = self.client.get(
+                f"/api/v1/dags/example_python_operator/dagRuns/TEST_DAG_RUN_ID/taskInstances/print_the_context/{map_index}",
```
Comment (ashb) [inline tests/api_connexion/endpoints/test_task_instance_endpoint.py]:
```suggestion
                "/api/v1/dags/example_python_operator/dagRuns/TEST_DAG_RUN_ID/taskInstances"
                + f"/print_the_context/{map_index}",
```

Comment (dstandish) [reply tests/api_connexion/endpoints/test_task_instance_endpoint.py]: thanks

**Pull request merged**